### ISSUE TITLE
Default headline for DatePicker displays incorrect date on iOS

### ISSUE THREAD

**Issue by j-roskopf** [JetBrains/compose-multiplatform#3903]
**Describe the bug**
When using the DatePicker component with the default headline, the date displayed is incorrect on iOS, but displays fine for me on Android

**Affected platforms**
- iOS 

**Versions**
- Kotlin version*: 1.9.20
- Compose Multiplatform version*: 1.5.10
- OS version(s)* (required for Desktop and iOS issues): iOS 17.0.1
- OS architecture (x86 or arm64): arm64
- JDK (for desktop issues): 

**To Reproduce**
Steps and/or the code snippet to reproduce the behavior:
1. Declare DatePicker
```
            val state = rememberDatePickerState()

            DatePicker(
                state = state,
                title = null,
            )
```
2. Select date
3. Observe headline value

**Expected behavior**
Value should be correct / the same as what displays on Android

**Screenshots**

Left is Android and right is iOS

https://github.com/JetBrains/compose-multiplatform/assets/7951665/df316488-0faa-4ed0-a9b8-86d6945af3f1

both devices are set to the same time + date

<img width="863" alt="Screenshot 2023-11-05 at 8 52 56 PM" src="https://github.com/JetBrains/compose-multiplatform/assets/7951665/ebf93e75-3cac-4960-bf5c-c821995b18f0">

**Comment by j-roskopf** [comment]
I am also now noticing in the demo video above, the date for the iOS date picker is labeled as October 2023, but it is November 2023

**Comment by alexzhirkevich** [comment]
It was already fixed in https://github.com/JetBrains/compose-multiplatform-core/pull/877 but sadly wasn't landed in latest release

**Comment by j-roskopf** [comment]
ah gotchya, I'll close the issue. Is there an estimated version it would ship in that I can be on the lookout for?

Thanks!

**Comment by MatkovIvan** [comment]
This will be included in 1.6.0 (even alpha's)

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-3903">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
